ocaml/xenopsd/xc/xenctrlext.mli

```diff
@@ -78,3 +78,33 @@ external vcpu_setaffinity_soft : handle -> domid -> int -> bool array -> unit
 external numainfo : handle -> numainfo = "stub_xenctrlext_numainfo"
 
 external cputopoinfo : handle -> cputopo array = "stub_xenctrlext_cputopoinfo"
+
+module Xenforeignmemory : sig
+  type handle
+
+  type mapping =
+    (char, Bigarray.int8_unsigned_elt, Bigarray.c_layout) Bigarray.Array1.t
```

Comment:
Could this be made abstract?

Reply:
I think the with_mapping already limits the lifetime of the bindings for this type and I don't know yet exactly how are the bytes going to be read. I'd rather keep it like it is now before restricting its usage